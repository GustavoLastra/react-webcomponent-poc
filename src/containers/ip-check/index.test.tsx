const mockIPCheckSelector = jest.fn();
jest.mock('./selector', () => ({
  __esModule: true,
  ipCheckSelector: mockIPCheckSelector
}));

import React from 'react';
import ReactTestRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import IPCheck from './index';

describe('<IPCheck />', () => {
  const mockStore = configureStore([]);

  it('should render and match the snapshot', () => {
    mockIPCheckSelector.mockReturnValue({ loading: true, status: { ip: '127.0.0.1', success: true, time: 173 } });
    // @ts-ignore
    const component = <IPCheck store={mockStore()} />;
    const renderer = ReactTestRenderer.createRenderer();
    renderer.render(component);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
