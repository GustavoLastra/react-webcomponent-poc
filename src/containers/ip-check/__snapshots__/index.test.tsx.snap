// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IPCheck /> should render and match the snapshot 1`] = `
<Context.Provider
  value={null}
>
  <IPCheck
    loading={true}
    status={
      Object {
        "ip": "127.0.0.1",
        "success": true,
        "time": 173,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</Context.Provider>
`;
